ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -173,9 +173,14 @@
       lastModifiedTimeSec = new AtomicLong(0);
       if (shouldMap) {
         map(false);
+        bloomFile = new File(indexFile.getParent(), indexSegmentFilenamePrefix + BLOOM_FILE_NAME_SUFFIX);
+        if (!bloomFile.exists()) {
+          bloomFilter = FilterFactory.getFilter(config.storeIndexMaxNumberOfInmemElements,
```

Comment:
i don't think we should use `config.storeIndexMaxNumberOfInmemElements` but use the actual number of keys in this segment. I think we should instantiate  this in the `generateBloomFileAndPersist()` function after we do the `getEntriesSince()` using the size of `indexEntries`.

Reply:
good point. Changes have been made.